src/main/java/io/cryostat/agent/remote/InvokeContext.java

```diff
@@ -0,0 +1,110 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.agent.remote;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class InvokeContext implements RemoteContext {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private final ObjectMapper mapper;
+
+    @Inject
+    InvokeContext(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @Override
+    public String path() {
+        return "/mbean-invoke/";
+    }
+
+    /*
+    * public <T> T invokeMBeanOperation(
+           String beanName,
+           String operation,
+           Object[] parameters,
+           String[] signature,
+           Class<T> returnType)
+    */
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            String mtd = exchange.getRequestMethod();
+            switch (mtd) {
+                case "POST":
+                    try (InputStream body = exchange.getRequestBody()) {
+                        MBeanInvocationRequest req =
+                                mapper.readValue(body, MBeanInvocationRequest.class);
+                        if (req.beanName.equals(ManagementFactory.MEMORY_MXBEAN_NAME)) {
+                            MemoryMXBean bean = ManagementFactory.getMemoryMXBean();
+                            bean.gc();
+                            exchange.sendResponseHeaders(HttpStatus.SC_OK, BODY_LENGTH_NONE);
+                        }
+                    } catch (Exception e) {
+                        log.error("mbean serialization failure", e);
+                        exchange.sendResponseHeaders(HttpStatus.SC_BAD_GATEWAY, BODY_LENGTH_NONE);
+                    }
+                    break;
+                default:
+                    log.warn("Unknown request method {}", mtd);
+                    exchange.sendResponseHeaders(
+                            HttpStatus.SC_METHOD_NOT_ALLOWED, BODY_LENGTH_NONE);
+                    break;
+            }
+        } finally {
+            exchange.close();
+        }
+    }
+
+    static class MBeanInvocationRequest<T> {
+
+        public String beanName;
+        public String operation;
+        public Object[] parameters;
+        public String[] signature;
+        public Class<T> returnType;
```

Comment:
I don't think the `returnType` is needed on this side - it's used on the Cryostat server side so that the internal callers can indicate how the response body should be parsed, but here on the Agent side I think that if the M(X)Bean operation being invoked has any return value, then we should just serialize that out as JSON or whatever format is applicable. If we want there to be some way for the server to request a particular format then that should primarily be done using HTTP content negotiation, ie the `Accept` request header.